tests/rancher/rancher_k3s_master.pm

```diff
@@ -0,0 +1,60 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# Summary: Web browser UI test for rancher container
+# Maintainer: George Gkioulis <ggkioulis@suse.com>
+
+use base 'x11test';
+use strict;
+use warnings;
+use testapi;
+use lockapi;
+use utils;
+use mm_network;
+use rancher::utils;
+use Utils::Systemd 'disable_and_stop_service';
+
+sub run {
+    my ($self) = @_;
+    $self->select_serial_terminal;
+
+    systemctl('start sshd.service');
+    disable_and_stop_service('firewalld');
+
+    mutex_wait('support_server_ready');
+    configure_hostname(get_required_var('HOSTNAME'));
+    configure_dhcp();
+
+    script_retry("nslookup gitlab.com", delay => 3, retry => 10);
+
+    assert_script_run("curl -L https://get.k3s.io -o ~/get_k3s", 360);
+    assert_script_run("chmod +rx ~/get_k3s");
+
+    assert_script_run("./get_k3s", 600);
+    script_retry("kubectl get node | grep ' Ready '", delay => 3, retry => 10);
+    script_run("cat /var/lib/rancher/k3s/server/node-token");
+
+    barrier_wait('k3s_prepared');
+
+    systemctl('status k3s');
+    
+    script_retry("[ `kubectl get node | grep ' Ready ' | wc -l` -eq 3 ]", delay => 15, retry => 40);
```

Comment:
I guess -eq 3 is because we have 1 master and two workers, isn't it?

Reply:
Yes, I'll create a comment there.